csharp/ql/test/experimental/ir/ir/PrintAst.expected

```diff
@@ -5,7 +5,7 @@ array.cs:
 #    3|     4: [BlockStmt] {...}
 #    4|       0: [LocalVariableDeclStmt] ... ...;
 #    4|         0: [LocalVariableDeclAndInitExpr] Int32[] one_dim = ...
-#    4|           -1: [TypeMention] Int32[]
+#    4|           -1: [TypeMention] int[]
 #    4|             1: [TypeMention] int
```

Comment:
@hvitved The main motivation for this commit was to unify the type name printing of `TypeMention`s. I think this looks a bit better, but we're still showing `Int32[]` in the above `LocalVariableDeclAndInitExpr` (and in some other nodes too). Should/Can I change those classes to use the `Type.toStringWithTypes`? It feels a bit strange to change all these base class `toString()`s just for the sake of AST printing.

What was the initial idea behind `toString` vs `toStringWithTypes`?